usecases/byteops/byteops.go

```diff
@@ -191,38 +192,102 @@ func (bo *ReadWriter) WriteByte(b byte) {
 	bo.Position += 1
 }
 
-func Float32ToByteVector(vec []float32) []byte {
-	byteVec := make([]byte, len(vec)*uint32Len)
-	for i := 0; i < len(vec); i++ {
-		binary.LittleEndian.PutUint32(byteVec[i*uint32Len:i*uint32Len+uint32Len], math.Float32bits(vec[i]))
+func Float32ToByteVector(floats []float32) []byte {
+	vector := make([]byte, len(floats)*uint32Len)
+	for i := 0; i < len(floats); i++ {
+		binary.LittleEndian.PutUint32(vector[i*uint32Len:(i+1)*uint32Len], math.Float32bits(floats[i]))
 	}
-	return byteVec
+	return vector
+}
+
+func Float64ToByteVector(floats []float64) []byte {
+	vector := make([]byte, len(floats)*uint64Len)
+	for i := 0; i < len(floats); i++ {
+		binary.LittleEndian.PutUint64(vector[i*uint64Len:(i+1)*uint64Len], math.Float64bits(floats[i]))
+	}
+	return vector
 }
 
-func Float64ToByteVector(vec []float64) []byte {
-	byteVec := make([]byte, len(vec)*uint64Len)
-	for i := 0; i < len(vec); i++ {
-		binary.LittleEndian.PutUint64(byteVec[i*uint64Len:i*uint64Len+uint64Len], math.Float64bits(vec[i]))
+func Float32FromByteVector(vector []byte) []float32 {
+	floats := make([]float32, len(vector)/uint32Len)
+
+	for i := 0; i < len(floats); i++ {
+		asUint := binary.LittleEndian.Uint32(vector[i*uint32Len : (i+1)*uint32Len])
+		floats[i] = math.Float32frombits(asUint)
 	}
-	return byteVec
+	return floats
 }
 
-func Float32FromByteVector(vecByte []byte) []float32 {
-	vector := make([]float32, len(vecByte)/uint32Len)
+func Float64FromByteVector(vector []byte) []float64 {
+	floats := make([]float64, len(vector)/uint64Len)
 
-	for i := 0; i < len(vector); i++ {
-		asUint := binary.LittleEndian.Uint32(vecByte[i*uint32Len : i*uint32Len+uint32Len])
-		vector[i] = math.Float32frombits(asUint)
+	for i := 0; i < len(floats); i++ {
+		asUint := binary.LittleEndian.Uint64(vector[i*uint64Len : (i+1)*uint64Len])
+		floats[i] = math.Float64frombits(asUint)
+	}
+	return floats
+}
+
+func BoolsToByteVector(bools []bool) []byte {
+	vector := make([]byte, len(bools))
+	for i, val := range bools {
+		if val {
+			vector[i] = 1
+		} else {
+			vector[i] = 0
+		}
 	}
 	return vector
 }
 
-func Float64FromByteVector(vecByte []byte) []float64 {
-	vector := make([]float64, len(vecByte)/uint64Len)
+func BoolsFromByteVector(vector []byte) []bool {
+	bools := make([]bool, len(vector))
+	for i, val := range vector {
+		if val == 1 {
+			bools[i] = true
+		} else {
+			bools[i] = false
+		}
+	}
+	return bools
+}
 
-	for i := 0; i < len(vector); i++ {
-		asUint := binary.LittleEndian.Uint64(vecByte[i*uint64Len : i*uint64Len+uint64Len])
-		vector[i] = math.Float64frombits(asUint)
+func StringsToByteVector(strings []string) []byte {
+	var vector []byte
+	delimiter := []byte{','}
```

Comment:
what if we have a `,` in the string?

Reply:
I rather assumed this would be forbidden but users might be storing entire paragraphs in fields so you're definitely right. I'll use the string lengths for decoding rather than this delimiter as I can't think of a good one that will be safe in all cases

Reply:
Have reverted to `repeated string values` rather than `bytes values` as below

Reply:
Alternative idea for strings, but it is totally fine to not do it:
Encode it as [][]byte?

Reply:
Hmm, that is a much better idea. I'll look into it

Reply:
I'm going to leave the `bool` and `string` values since encoding/decoding them is more involved than for `float64` values. If we have time before the next release we could add it in with a benchmark too